src/dbnode/persist/fs/commitlog/reader.go

```diff
@@ -207,9 +205,64 @@ func (r *reader) readLogEntry() error {
 	return nil
 }
 
+func (r *reader) namespaceIDReused(id []byte) ident.ID {
+	var namespaceID ident.ID
+	for _, ns := range r.namespacesRead {
```

Comment:
Yeah, also would cause a map lookup every single read which has shown with small numbers of elements to be much slower than looping and checking for bytes comparison.